### ISSUE TITLE
"tfci run create" gets stuck after successful apply

### ISSUE THREAD

**Issue by qrkourier** [hashicorp/tfc-workflows-tooling#15]
This issue applies to a build I created from branch `main` on commit `f0abf3b740ca3635db3ea03e75215d03bb6456d4` ([commit](https://github.com/hashicorp/tfc-workflows-tooling/commit/f0abf3b740ca3635db3ea03e75215d03bb6456d4)).

I have a workspace set to auto-apply. I used `tfci run create` to trigger a run. The plan was applied, but `tfci` got stuck.

I built the container like this.

```bash
docker buildx build -t kentest . --load
```

I ran it like this.

```bash
docker run -it --rm \
  -e "TF_API_TOKEN" \
  -e "TF_CLOUD_ORGANIZATION" \
  kentest \
  tfci run create -workspace myworkspace
```

The symptom was that stdin was never released, and `tfci` kept on printing:

```
Run Status: 'applied'
Run Status: 'applied'
Run Status: 'applied'
Run Status: 'applied'
Run Status: 'applied'
Run Status: 'applied'
```

...for 1/2 hour after the successful apply showed completed in TF Cloud. I eventually pressed Ctrl-C to release stdin. The workspace had already been unlocked and other runs triggered.

**Comment by qrkourier** [comment]
Here's the output from when I sent a SIGQUIT to the `tfci` process while reproducing this issue.

[tfci-sigquit.txt](https://github.com/hashicorp/tfc-workflows-tooling/files/11681215/tfci-sigquit.txt)

**Comment by ggambetti** [comment]
This appears to be a concurrency issues with the way [runs are processed for creation](https://github.com/hashicorp/tfc-workflows-tooling/blob/main/internal/cloud/run.go#L154-L164): `applied` is not accounted for. It could occur before the tool notices the `run-planned` state, especially for workspaces configured with auto-apply.

**Comment by mjyocca** [comment]
Fixed as of #16
